src/utilities/core-utils.test.ts

```diff
@@ -91,6 +91,126 @@ describe("CoreUtils", () => {
+        test("when given numeric enum, returns an object of values", () => {
+            // Arrange
+            enum testEnum {
+                first,
+                second,
+                third,
+            }
+
+            // Act
+            const result = CoreUtils.numericEnumToPojo(testEnum);
+            const values = CoreUtils.objectToArray(result);
+
+            // Assert
+            expect(values).toContain(testEnum.first);
+            expect(values).toContain(testEnum.second);
+            expect(values).toContain(testEnum.third);
+            expect(values).toHaveLength(3);
+        });
+
+        test("when given non numeric enum, returns object of values", () => {
+            // Arrange
+            enum testEnum {
+                first = "First", second = "Second",
+            }
+
+            // Act
+            const result = CoreUtils.numericEnumToPojo(testEnum);
+            const values = CoreUtils.objectToArray(result);
+
+            // Assert
+            expect(values).toContain(testEnum.first);
+            expect(values).toContain(testEnum.second);
+            expect(values).toHaveLength(2);
+        });
+
+        test("when given null enum, returns empty object", () => {
+            // Act
+            const result = CoreUtils.numericEnumToPojo(null);
+
+            // Assert
+            expect(result).toMatchObject({});
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion numericEnumToPojo
+
+    // -------------------------------------------------------------------------------------------------
+    // #region ObjectToArray
+    // -------------------------------------------------------------------------------------------------
+
+    describe("objectToArray", () => {
+        test("convert object to array of values", () => {
+            // Arrange
+            const testObject = {
+                name: "Jane",
+                age:  30
+            }
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toContain(testObject.name);
+            expect(result).toContain(testObject.age);
+            expect(result).toHaveLength(2);
+        });
+
+
+        test("given an empty object, return an empty array", () => {
+            // Arrange
+            const testObject = {};
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion ObjectToArray
+
+    // -------------------------------------------------------------------------------------------------
+    // #region sleep
+    // -------------------------------------------------------------------------------------------------
+
+    describe("sleep", () => {
+        test("schedule n milliseconds", () => {
+            // Arrange
+            jest.useFakeTimers();
```

Comment:
what's this? I prob could have used this for my tests elsewhere where I'm using `SleepSync` lol

Reply:
jest has an some built-in options to work around sleep where it lets you create fake timer and jump ahead of n milliseconds to test if certain things has happened. Pretty neat